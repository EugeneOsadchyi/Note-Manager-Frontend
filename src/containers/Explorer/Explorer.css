.Explorer {
  border: 2px solid black;
  width: 300px;
  height: 100%;
  overflow: scroll;
}
