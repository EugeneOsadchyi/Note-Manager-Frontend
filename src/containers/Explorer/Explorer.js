import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import './Explorer.css';

import DirectoryList from '../../components/DirectoryList';
import * as actions from '../../store/actions/directory';

export const Explorer = (props) => {
  const {
    directories,
    onSelectDirectory,
    onOpenDirectory,
    onCloseDirectory,
  } = props;

  return (
    <div className="Explorer">
      <DirectoryList
        directories={directories}
        selectDirectory={onSelectDirectory}
        openDirectory={onOpenDirectory}
        closeDirectory={onCloseDirectory}
      />
    </div>
  );
};

Explorer.propTypes = {
  directories: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.name,
      parentId: PropTypes.number,
      opened: PropTypes.bool,
    }),
  ).isRequired,
  onSelectDirectory: PropTypes.func.isRequired,
  onOpenDirectory: PropTypes.func.isRequired,
  onCloseDirectory: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  directories: state.directory.directories,
});

const mapDispatchToProps = dispatch => ({
  onSelectDirectory: id => dispatch(actions.selectDirectory(id)),
  onOpenDirectory: id => dispatch(actions.openDirectory(id)),
  onCloseDirectory: id => dispatch(actions.closeDirectory(id)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Explorer);
