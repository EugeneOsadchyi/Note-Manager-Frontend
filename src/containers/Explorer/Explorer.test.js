import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { Explorer } from './Explorer';

describe('Explorer', () => {
  const mockOnSelectDirectory = jest.fn();
  const mockOnOpenDirectory = jest.fn();
  const mockOnCloseDirectory = jest.fn();

  const defaultProps = {
    directories: [
      {
        id: 1, name: 'Root', parentId: null, opened: true, active: true,
      }, {
        id: 2, name: 'Child', parentId: 1, opened: false, active: false,
      },
    ],
    onSelectDirectory: mockOnSelectDirectory,
    onOpenDirectory: mockOnOpenDirectory,
    onCloseDirectory: mockOnCloseDirectory,
  };

  const explorer = shallow(<Explorer {...defaultProps} />);

  it('renders properly', () => {
    expect(shallowToJson(explorer)).toMatchSnapshot();
  });

  it('contains DirectoryList component', () => {
    expect(explorer.find('DirectoryList').exists()).toBe(true);
  });

  it('fulfils DirectoryList component with required props', () => {
    const expectedProps = {
      directories: [...defaultProps.directories],
      parentId: undefined,
      selectDirectory: defaultProps.onSelectDirectory,
      openDirectory: defaultProps.onOpenDirectory,
      closeDirectory: defaultProps.onCloseDirectory,
    };
    expect(explorer.find('DirectoryList').props()).toEqual(expectedProps);
  });
});
