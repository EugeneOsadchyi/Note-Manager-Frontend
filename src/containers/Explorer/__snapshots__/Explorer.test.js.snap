// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Explorer renders properly 1`] = `
<div
  className="Explorer"
>
  <DirectoryList
    closeDirectory={[Function]}
    directories={
      Array [
        Object {
          "active": true,
          "id": 1,
          "name": "Root",
          "opened": true,
          "parentId": null,
        },
        Object {
          "active": false,
          "id": 2,
          "name": "Child",
          "opened": false,
          "parentId": 1,
        },
      ]
    }
    openDirectory={[Function]}
    selectDirectory={[Function]}
  />
</div>
`;
